sdk/monitor/monitor-opentelemetry-exporter/src/export/statsbeat/longIntervalStatsbeatMetrics.ts

```diff
@@ -0,0 +1,286 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+import {
+  createDefaultHttpClient,
+  createPipelineRequest,
+  HttpMethods,
+} from "@azure/core-rest-pipeline";
+import {
+  diag,
+  BatchObservableResult,
+  ObservableGauge,
+  ObservableResult,
+  Meter,
+} from "@opentelemetry/api";
+import { ExportResultCode } from "@opentelemetry/core";
+import {
+  MeterProvider,
+  PeriodicExportingMetricReader,
+  PeriodicExportingMetricReaderOptions,
+} from "@opentelemetry/sdk-metrics";
+import { AzureMonitorExporterOptions, AzureMonitorStatsbeatExporter } from "../../index";
+import * as ai from "../../utils/constants/applicationinsights";
+import {
+  StatsbeatCounter,
+  StatsbeatResourceProvider,
+  STATSBEAT_LANGUAGE,
+  AIMS_URI,
+  AIMS_API_VERSION,
+  AIMS_FORMAT,
+  EU_CONNECTION_STRING,
+  EU_ENDPOINTS,
+  NON_EU_CONNECTION_STRING,
+  CommonStatsbeatProperties,
+  VirtualMachineInfo,
+  AttachStatsbeatProperties,
+  StatsbeatFeatureType,
+  StatsbeatOptions,
+} from "./types";
+
+let instance: LongIntervalStatsbeatMetrics | null = null;
+
+const os = require("os");
+
+/**
+ * Long Interval Statsbeat Metrics
+ * @internal
+ */
+class LongIntervalStatsbeatMetrics {
+  private _AZURE_MONITOR_STATSBEAT_FEATURES = process.env.AZURE_MONITOR_STATSBEAT_FEATURES;
+  private _statsCollectionLongInterval: number = 86400000; // 1 day
+  private _isInitialized: boolean = false;
+
+  // Custom dimensions
+  private _vmInfo: VirtualMachineInfo = {};
+  private _resourceIdentifier: string = "";
+  private _resourceProvider: string = StatsbeatResourceProvider.unknown;
+  private _os: string = os.type();
+  private _cikey: string;
+  private _runtimeVersion: string;
+  private _language: string;
+  private _version: string;
+  private _attach: string = "sdk";
+
+  private _commonProperties: CommonStatsbeatProperties;
+  private _attachProperties: AttachStatsbeatProperties;
+
+  private _feature: number = 0;
+  private _instrumentation: number = 0;
+
+  private _longIntervalStatsbeatMeterProvider: MeterProvider;
+  private _longIntervalAzureExporter: AzureMonitorStatsbeatExporter;
+  private _longIntervalMetricReader: PeriodicExportingMetricReader;
+  private _longIntervalStatsbeatMeter: Meter;
+
+  // Network Attributes
+  private _connectionString: string;
+
+  // Observable Gauges
+  private _featureStatsbeatGauge: ObservableGauge;
+  private _attachStatsbeatGauge: ObservableGauge;
+
+  constructor(options: StatsbeatOptions) {
+    this._connectionString = this._getConnectionString(options.endpointUrl);
+    const exporterConfig: AzureMonitorExporterOptions = {
+      connectionString: this._connectionString,
+    };
+
+    if (this._AZURE_MONITOR_STATSBEAT_FEATURES) {
+      this._feature = JSON.parse(this._AZURE_MONITOR_STATSBEAT_FEATURES).feature;
```

Comment:
JSON parse should be always wrapped by try catch to avoid wrong values and prevent crashes